### PR TITLE
context added to request

### PR DESCRIPTION
**Pull request by gozeloglu**
* Fixed https://github.com/binalyze/httpreq/issues/11 issue.
* Tests are refactored with new `New()` function syntax.
* Unit tests added.

### CODE REVIEW

File: request.go [hunk 1]
```diff
@@ -35,6 +36,15 @@ func New(address string) *Req {
 		Method: "GET",
 		Header: make(http.Header),
 	}
+	if ctx != nil {
```
Comment (yakuter) [inline request.go:39]: Well the following usage is a better practice. Lets change it this way:

```
if ctx == nil {
	ctx = context.Background()
}

r.request = (&http.Request{
	Method: "GET",
	Header: make(http.Header),
}).WithContext(ctx)
```

File: request_test.go [hunk 2]
```diff
@@ -61,7 +62,7 @@ func TestMethods(t *testing.T) {
 
 	for _, row := range table {
 		url := fmt.Sprintf("%s/"+row.name, server.URL)
-		r := New(url)
+		r := New(context.TODO(), url)
```
Comment (yakuter) [inline request_test.go:65]: I see that you used context.TODO(). Please change them with context. Background(). I know that in codebase they are exactly same however the intent of the usage is different. We only have one context, no confusion and we use this for tests. So context.Background() is the suitable one.

For more explanation: https://www.programming-books.io/essential/go/context-todo-vs-context-background-d5224e27ff724a33a79cb4e03a5eb333

**Review by yakuter — CHANGES_REQUESTED** [review]

File: request.go (hunk 1 above)
Comment (gozeloglu) [reply request.go]: Fixed in https://github.com/binalyze/httpreq/pull/12/commits/afbc08a0ae51ce5a56639f7906ecfc65e0f18886

File: request_test.go (hunk 2 above)
Comment (gozeloglu) [reply request_test.go]: Fixed in https://github.com/binalyze/httpreq/pull/12/commits/afbc08a0ae51ce5a56639f7906ecfc65e0f18886.

**Review by yakuter — CHANGES_REQUESTED** [review]
Current changes are OK. However our main object to add context is make requests cancellable. So can you write test to see if we can cancel an ongoing request with `context.WithCancel` or `context.WithTimeout`

**Comment by gozeloglu** [comment]
> Current changes are OK. However our main object to add context is make requests cancellable. So can you write test to see if we can cancel an ongoing request with `context.WithCancel` or `context.WithTimeout`

I added unit test in https://github.com/binalyze/httpreq/pull/12/commits/cb54bbce5e57ed78590fa16738fb1a4c04c53c72. I hope I got it right and the unit test is correct.

**Review by yakuter — APPROVED** [review]

**Comment by yakuter** [comment]
Thank you @gozeloglu for this great job. If I have time, I will add a test with real http server and request to test the cancellation.

**Pull request merged**